interfaces/builtin/pwm_control.go

```diff
@@ -0,0 +1,61 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Zygmunt Krynicki
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+// https://www.kernel.org/doc/Documentation/pwm.txt
+const pwmControlSummary = `allows control of all aspects of PWM channels`
+
+// Controlling all aspects of PWM channels can potentially impact other snaps
+// and grant wide access to specific hardware and the system, so treat as
+// super-privileged
+const pwmControlBaseDeclarationPlugs = `
+  pwm-control:
+    allow-installation: false
+    deny-auto-connection: true
+`
+const pwmControlBaseDeclarationSlots = `
+  pwm-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const pwmControlConnectedPlugAppArmor = `
+# Description: Allow controlling all aspects of PWM channels. This can
+# potentially impact the system and other snaps, and allows privileged access
+# to hardware.
+
+/sys/class/pwm/pwmchip[0-9]*/{,un}export rw,
+/sys/class/pwm/pwmchip[0-9]*/npwm r,
+/sys/class/pwm/pwmchip[0-9]*/pwm[0-9]*/{period,duty_cycle,polarity,enable} rw,
```

Comment:
afaict pwm itself, used `/*` for the last component here?

Reply:
Yes, I can adjust that. I just wanted to be precise, as it is easier to extend but harder to take the permissions away.

If you want I can change that to the general glob.

Reply:
I agree, but given the choice for pwm itself, I think it's better to be consistent at this point, also it would be strange for the `-control` variant to be more restricted in this way than the non-control one

Reply:
Ack, I'll send an update in a moment.

Reply:
@pedronis done. I have also added the `k` permission to match the `pwm` interface better (`k` allows locking AFAIR).